sunpy/net/dataretriever/sources/goes.py

```diff
@@ -168,9 +177,163 @@ def _can_handle_query(cls, *query):
+
+        suvi_operational = {
+            16: TimeRange("2018-06-01", parse_time("now")),
+        }
+
+        results = []
+        for sat_num in suvi_operational:
+            if date in suvi_operational[sat_num]:
+                # if true then the satellite with sat_num is available
+                results.append(sat_num)
+
+        if results:
+            # Return the newest satellite
+            return max(results)
+        else:
+            # if no satellites were found then raise an exception
+            raise ValueError(f"No operational SUVI instrument on {date.strftime(TIME_FORMAT)}")
+
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+        Returns a URL to the SUVI data for the specified time range.
+
+        Parameters
+        ----------
+        timerange: sunpy.time.TimeRange
+            time range for which data is to be downloaded.
+        level : str
+            The level of the data.
+        wavelength :
+            Wavelength band.
+        satellitenumber : int
+            GOES satellite number. Must be >= 16.
+
+        """
+        base_url = "https://data.ngdc.noaa.gov/platforms/solar-space-observing-satellites/goes/goes{goes_number}/"
+        supported_waves = [94, 131, 171, 195, 284, 304]
+        supported_levels = ("2", "1b")
+
+        # these are optional requirements so if not provided assume defaults
+        # if wavelength is not provided assuming all of them
+        if "wavelength" in kwargs.keys():
+            wavelength_input = kwargs.get("wavelength")
+            if isinstance(wavelength_input, u.Quantity):  # not a range
+                if int(wavelength_input.to_value('Angstrom')) not in supported_waves:
+                    raise ValueError(f"Wavelength {kwargs.get('wavelength')} not supported.")
+                else:
+                    wavelength = [kwargs.get("wavelength")]
+            else:  #  _Range was provided
+                compress_index = [wavelength_input.wavemin <= this_wave <= wavelength_input.wavemax for this_wave in (supported_waves * u.Angstrom)]
+                if not any(compress_index):
+                    raise ValueError(
+                        f"Wavelength {wavelength_input} not supported.")
+                else:
+                    wavelength = list(compress(supported_waves, compress_index)) * u.Angstrom
+        else:  # no wavelength provided return all of them
+            wavelength = supported_waves * u.Angstrom
+        # check that the input wavelength can be converted to angstrom
+        waves = [int(this_wave.to_value('angstrom', equivalencies=u.spectral())) for this_wave in wavelength]
+        # use the given satellite number or choose the best one
+        satellitenumber = int(kwargs.get("satellitenumber", self._get_goes_sat_num(timerange.start)))
+        if satellitenumber < 16:
+            raise ValueError(f"Satellite number {satellitenumber} not supported.")
```

Comment:
Same potentially applies here, at least for consistency, although it would be a lot harder to come up with a breaking query.

Reply:
This is consistent with how the `XRSClient` does it.